Test Willy

WillyReyno opened the pull request.

WillyReyno committed:
end mouf test

WillyReyno committed:
fixing edit form

nguyenk commented on src/MoufTest/Controllers/CarController.php:
```diff
@@ -0,0 +1,216 @@
+     */
+    public function index($page = 1)
+    {
+        $webLibraryManager = $this->template->getWebLibraryManager();
```
l'injection de weblibraries peut se faire directement dans Mouf, c'est recommandé lorsqu'il s'agit de weblibrairies globales

WillyReyno replied:
C'est bon à savoir merci !

nguyenk commented on src/MoufTest/Controllers/CarController.php:
```diff
@@ -0,0 +1,216 @@
+        $webLibraryManager->addJsFile('src/public/js/livesearch.min.js');
+
+        // Define cars showed by page
+        $cars_per_page = 4;
```
Il faudrait utiliser une variable de configuration ou une constante

WillyReyno replied:
C'est fait !

nguyenk commented on src/MoufTest/Controllers/CarController.php:
```diff
@@ -0,0 +1,216 @@
+        $webLibraryManager->addJsFile('src/public/js/jquery.validate.min.js');
+        $webLibraryManager->addJsFile('src/public/js/form-validation.js');
+
+        $brands = $this->daoFactory->getBrandDao()->findAll()->jsonSerialize();
```
Pourquoi utiliser ```->jsonSerialize()``` ? Tu peux manipuler des objets dans TWIG directement

WillyReyno replied:
En effet, c'est inutile, j'ai retiré !

nguyenk commented on src/MoufTest/Controllers/CarController.php:
```diff
@@ -0,0 +1,216 @@
+     * @param int $id
+     * @return HtmlResponse
+     */
+    public function edit($id)
```
Il faut mutualiser les codes d'affichage du formulaire (autrement dit les URLs ```car/{id}/edit```et ```car/create```) et de sauvegarde (```car/update``` et ```car/store```)

WillyReyno replied:
C'est fait !

nguyenk commented on src/MoufTest/Controllers/CarController.php:
```diff
@@ -0,0 +1,216 @@
+        $carBean = new CarBean($brand, $name, $max_speed);
+        $this->daoFactory->getCarDao()->save($carBean);
+
+        Return new RedirectResponse('http://localhost/mouf-tests/car/1');
```
attention a la case : **r**eturn au lieu de **R**eturn, et l'url de retour doit être soit la liste des voitures soit le formulaire de la voiture créé. Dans tous les cas, il faut un message de confirmation (essaye de faire marcher le [User Message Service](https://github.com/thecodingmachine/html.widgets.messageservice))

WillyReyno replied:
C'est corrigé pour la case. 

Par contre, j'ai essayé de faire marcher le User Message Service, mais il ne m'affiche les messages que dans l'UI de Mouf et non pas sur mes vues en front, je ne sais pas si c'est normal ?

nguyenk commented on views/car/index.twig:
```diff
@@ -0,0 +1,78 @@
+
+        $("#cars-table").tablesorter();
+
+        $(".delete-car").click(function () {
```
traiter le retour ajax : il faut sans doute recharger la liste non ?

WillyReyno committed:
- Made a configuration variable for cars per page
- Removed jsonSerialize from brand calls
- Fixed typo Return -> return
- Merged create and edit methods in CarController
- Merged store and update methods in CarController

WillyReyno committed:
revert readme
